torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -483,12 +497,12 @@ def record_pre_forward(self, handles: List[FlatParamHandle]) -> None:
         handles_key = tuple(handles)
         if self._checking_order:
             self._check_order(handles_key)
-        # Fix the order after the first iteration and only record the first
-        # usage of a handles key
-        if (
-            not self.is_first_iter
-            or handles_key in self.handles_to_pre_forward_order_index
-        ):
+            # Fix the debug pre-forward order after the first iteration since
+            # we check against the first iteration order thereafter
+            if self.is_first_iter:
+                self.handles_debug_pre_forward_order.append(handles_key)
+        # Only record the first usage of a handles key
+        if handles_key in self.handles_to_post_forward_order_index:
```

Comment:
in 'handles_to_pre_forward_order_index'? why not record the order every iteration from the scratch? it is good for dynamic graph?